Automattic/wp-calypso
# Emails: track creating new email accounts by titan free trial users

#### Proposed Changes

In this PR, we add additional track information to distinguish clicks by titan free trial users and others.
So now we send the same `calypso_inbox_new_mailbox_upsell_click` event, but additionally we add parameter `context` as `paid` | `free`.

#### Testing Instructions
Let's check a few scenarios.
**1) Test receiving `context` as `free` for - titan FREE TRIAL users**
* Go to your dashboard
* Click on the "Switch Site" button in the top left corner
* If you have existed account with Professional Email (FREE TRIAL):
	* Choose your account with the Professional Email (FREE TRIAL)
* Otherwise, if you don't have existed account with the Professional Email (FREE TRIAL):
    * Click on "Add new site" at the end of the appeared list
    * Enter any domain name for test purposes
    * In the appeared list below - choose a domain with the help of the "Select" button
    * Choose any plan from the proposed with the help of the "Select" button
    * Scroll down and pay for the subscription with the help of the "Pay {amount} with credits" button
    * Fill data in the "Professional Email" form to create your 3-month free email account and click on the "Add professional Email" button
    * Scroll down and click on "Complete Checkout"
    * On the page "What are your goals?" - click on "Skip to dashboard" in the top right corner
* In the left sidebar menu, click on the "Inbox" item
* In the page body, click on the "Create a new mailbox" button

**ER (Expected result)**: in the [Tracks Vigilante](https://github.com/Automattic/tracks-chrome-extension) appeared event `calypso_inbox_new_mailbox_upsell_click` with parameter `context` as `free`. AND in some time the same information appeared in "Tracks" system.



**2) Test receiving `context` as `paid` for - titan PAID users**
* Move the mouse on the "Upgrades" item in the left sidebar
* In the dropdown, click on the "Emails" item
* Click on the "Renew now" button, which is located at the top part of the page body (on the right side of the text "Expires: {Date}")
* Scroll down and click on "Pay {amount} with credits"
* In the top left corner of the page, click the "My Sites" button
* In the left sidebar, click on the "Inbox" item
* In the page body, click on the "Create a new mailbox" button

**ER (Expected result)**: in the [Tracks Vigilante](https://github.com/Automattic/tracks-chrome-extension) appeared event `calypso_inbox_new_mailbox_upsell_click` with parameter `context` as `paid`. AND in some time the same information appeared in "Tracks" system.



**3) Test receiving `context` as `paid` for - Google workspace paid users**

* Go to your dashboard
* Click on the "Switch Site" button in the top left corner
* Recommended if you have existed account with "Google Workspace" subscription:
	* Choose your account with the "Google Workspace" subscription
* Otherwise, if you don't have existed account with the "Google Workspace" subscription:
    * Click on "Add new site" at the end of the appeared list
    * Enter any domain name for test purposes
    * In the appeared list below - choose a domain with the help of the "Select" button
    * Choose any plan from the proposed with the help of the "Select" button
    * Scroll down and pay for the subscription with the help of the "Pay {amount} with credits" button
    * Note, on the page "Add Professional Email" - click on the "Skip for now" button
    * On the page "What are your goals?" - click on "Skip to dashboard" in the top right corner
    * In the left sidebar menu, click on the "Inbox" item
    * At the bottom of the page is located the "Google Workspace" section, click on the "Select" button, on the right side of the section
    * Fill "Google Workspace" form and click on the "Purchase" button, below the form
    * Click on "Pay {amount} with credits"
    * Click on the "Manage email" button in the middle of the page body
    * In the left sidebar menu, click on the "Inbox" item
    * If you see the warning "Configuring mailboxes" at the top part of the page body - it means that it's necessary to wait a bit, so at the current moment, you can go to the kitchen and make a cup of coffee 🙃
    * Reload the page to check for any news regarding the configuration of your mailboxes. 
    * If it's finished, you will see the notification "Action required". So click on the "Finish setup" button under the notification and follow the instructions.
	* After finishing setup - come back to the dashboard
* In the left sidebar, click on the "Inbox" item
* In the page body, click on the "Create a new mailbox" button

**ER (Expected result)**: in the [Tracks Vigilante](https://github.com/Automattic/tracks-chrome-extension) appeared event `calypso_inbox_new_mailbox_upsell_click` with parameter `context` as `paid`. AND in some time the same information appeared in "Tracks" system.

#### Pre-merge Checklist


Related to #
1200182182542585-as-1202735679392329/f
